tauri-utils/src/config.rs

```diff
@@ -6,16 +6,75 @@ use serde_json::Value as JsonValue;
 
 use std::collections::HashMap;
 
+/// The window webview URL options.
+#[derive(PartialEq, Debug, Clone)]
+pub enum WindowUrl {
+  /// The app's index URL.
+  App,
+  /// A custom URL.
+  Custom(String),
+}
+
+impl Default for WindowUrl {
+  fn default() -> Self {
+    Self::App
+  }
+}
+
+impl<'de> Deserialize<'de> for WindowUrl {
+  fn deserialize<D>(deserializer: D) -> Result<WindowUrl, D::Error>
+  where
+    D: Deserializer<'de>,
+  {
+    struct StringVisitor;
+    impl<'de> Visitor<'de> for StringVisitor {
+      type Value = WindowUrl;
+      fn expecting(&self, formatter: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        formatter.write_str("a string representing an url")
+      }
+
+      fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+      where
+        E: serde::de::Error,
+      {
+        if v.to_lowercase() == "app" {
+          Ok(WindowUrl::App)
+        } else {
+          Ok(WindowUrl::Custom(v.to_string()))
+        }
+      }
+    }
+    deserializer.deserialize_str(StringVisitor)
+  }
+}
+
 /// The window configuration object.
-#[derive(PartialEq, Deserialize, Debug)]
-#[serde(tag = "window", rename_all = "camelCase")]
+#[derive(PartialEq, Deserialize, Debug, Clone)]
 pub struct WindowConfig {
+  #[serde(default = "default_window_label")]
+  /// The window identifier.
+  pub label: String,
+  /// The window webview URL.
+  #[serde(default)]
+  pub url: WindowUrl,
+  /// The horizontal position of the window's top left corner
+  pub x: Option<f64>,
```

Comment:
I think we should accept `usize` for these and convert them to i64 on the wry side. I can't think of a scenario where you'd actually pass a non integer so it makes more sense to accept an integer imo, it will save people from having to add the `.0`, which could be confusing for those unfamiliar with Rust and statically typed languages.

Reply:
I was just following the `wry` interface. We should follow the same convention.
This is the `tauri.conf.json` stuff and the deserialization would also work with integers. The only real issue regarding this would be the Rust API to create windows, but I can also change that to accept both `usize` and `f64`.

Reply:
I think that Wry itself should take `usize`s, but if Tauri accepts either that's ok too.